src/ui/QuerySummary/QuerySummary.ts

```diff
@@ -189,19 +238,72 @@ export class QuerySummary extends Component {
     }
   }
 
+  // TODO : For now, the function is built such as the custom message can only contain one QUERY_TAG
+  //        Do we want to be able to put mutiple QUERY_TAG ?
+  private parseNoResultsFoundMessage(noResultsFoundMessage: string) {
+    let parsedNoResultsFoundMessage = noResultsFoundMessage.split(QUERY_TAG, 2);
+    if (parsedNoResultsFoundMessage.length == 1) {
+      parsedNoResultsFoundMessage.push('');
+    }
+    return parsedNoResultsFoundMessage;
+  }
+
+  private isDefautlNoResulstFoundMessage() {
+    return this.options.noResultsFoundMessage == DEFAULT_NO_RESULT_FOUND_MESSAGE;
+  }
+
+  private isSubArrayInArray(array: string, subarray: string) {
+    let subArrayfound: boolean;
+    let subArrayLength: number = subarray.length,
+      arrayLengthToScan: number = array.length + 1 - subArrayLength;
+
+    for (let i = 0; i < arrayLengthToScan; i++) {
+      subArrayfound = true;
+      for (let j = 0; j < subArrayLength; j++) {
+        if (array[i + j] !== subarray[j]) {
+          subArrayfound = false;
+          break;
+        }
+      }
+      if (subArrayfound) {
+        return true;
+      }
+    }
+    return false;
+  }
+
   private displayInfoOnNoResults() {
     const queryEscaped = escape(this.queryStateModel.get(QueryStateModel.attributesEnum.q));
-    let noResultsForString: Dom;
-
-    if (queryEscaped != '') {
-      noResultsForString = $$(
-        'div',
-        {
-          className: 'coveo-query-summary-no-results-string'
-        },
-        l('noResultFor', $$('span', { className: 'coveo-highlight' }, queryEscaped).el.outerHTML)
-      );
+    let queryEscapedValue: string;
+    // TODO : I didn't find an equivalent for isSubArrayInArray() but something might exist?
+    const isQueryTagInMessage = this.isSubArrayInArray(this.options.noResultsFoundMessage, QUERY_TAG);
+
+    if (!isQueryTagInMessage && !this.isDefautlNoResulstFoundMessage()) {
+      queryEscapedValue = '';
+    } else {
+      queryEscapedValue = queryEscaped;
     }
+
+    let queryEscapedString = $$(
```

Comment:
It would be cleaner if all the logic involved in preparing the query element could be moved into a separate method, maybe called `getQueryElement`.